userspace/falco/configuration.cpp

```diff
@@ -105,8 +102,101 @@ void falco_configuration::init(const std::string& conf_filename, const std::vect
 	load_yaml(conf_filename, config);
 }
 
+void falco_configuration::load_engine_config(const std::string& config_name, const yaml_helper& config)
+{
+	// Set driver mode if not already set.
+	const std::unordered_map<std::string, engine_kind_t> engine_mode_lut = {
+		{"kmod",engine_kind_t::KMOD},
+		{"ebpf",engine_kind_t::EBPF},
+		{"modern-ebpf",engine_kind_t::MODERN_EBPF},
+		{"replay",engine_kind_t::REPLAY},
+		{"gvisor",engine_kind_t::GVISOR},
+		{"none",engine_kind_t::NONE},
+	};
+
+	constexpr int16_t default_buf_size_preset = 4;
+	constexpr int16_t default_cpus_for_each_syscall_buffer = 2;
+	constexpr bool default_drop_failed_exit = false;
```

Comment:
These defaults are a bit hidden. However, not a big issue.

Reply:
Yep, they will go away in next release cycle.